modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -2182,7 +2182,9 @@ def applyier(df, internal_indices, other=[], internal_other_indices=[]):
     # __getitem__ methods
     __getitem_bool = Binary.register(
         # r is usually a list, but when r.size == 1, the array is squeezed to a scalar
-        lambda df, r: df[r] if r.size > 1 else df[[r]],
+        lambda df, r: df[r]
+        if r.size > 1 or isinstance(r, (pandas.Series, pandas.DataFrame)) and r.empty
+        else df[[r]],
```

Comment:
why do we even need to depend on `.size`? can we just use the `is_scalar` check?
```suggestion
        lambda df, r: df[[r]] if is_scalar(r) else df[r],
```

Reply:
done